qt/widgets/common/src/SequentialFitDialog.cpp

```diff
@@ -373,7 +372,61 @@ void SequentialFitDialog::functionChanged() { populateParameters(); }
 
 void SequentialFitDialog::finishHandle(
     const Mantid::API::IAlgorithm * /*alg*/) {
+  getFitResults();
   emit needShowPlot(&ui, m_fitBrowser);
+  m_fitBrowser->sequentialFitFinished();
+}
+
+/// Set the parameters to the fit outcome.
+/// if more than one entry in the gui the parameters used are first entry with
+/// same name as fit property browser if one entry will try and use row that
+/// corresponds to ws index in fit browser, if not found defaults to first row.
+void SequentialFitDialog::getFitResults() {
+  auto wsName = m_outputName;
+  if (Mantid::API::AnalysisDataService::Instance().doesExist(wsName)) {
+    Mantid::API::ITableWorkspace_sptr ws =
+        boost::dynamic_pointer_cast<Mantid::API::ITableWorkspace>(
+            Mantid::API::AnalysisDataService::Instance().retrieve(wsName));
```

Comment:
This dynamic cast is unchecked, could it ever fail in the future?

Reply:
It shouldn't happen as the workspace is made during the fit but I'll add a check in just in case someone changes the way the sequential fit algorithm works